unit_tests/testStrings/testStrings.f90

```diff
@@ -742,9 +742,9 @@ SUBROUTINE testStrFunct()
   str = '-0.2468e-000'
   ASSERT(str%isNumeric(),'with e-000')
   str = '-.2468e-000'
-  ASSERT(.NOT.str%isNumeric(),'no leading digit')
+  ASSERT(str%isNumeric(),'no leading digit')
   str = '-0.2468-000'
-  ASSERT(.NOT.str%isNumeric(),'mixing exponent symbol')
+  ASSERT(str%isNumeric(),'mixing exponent symbol')
```

Comment:
Is this actually correct?  The missing `e` makes that not really a single numeric... unless we're supposed to catch artihmetic expressions too...

Reply:
So I think this `10.7.2.3.2 F editing` in the standard applies 
> The input field is either an IEEE exceptional specification or consists of an optional sign, followed by a string of
 one or more digits optionally containing a decimal symbol, including any blanks interpreted as zeros. The d has
 no effect on input if the input field contains a decimal symbol. If the decimal symbol is omitted, the rightmost
 d digits of the string, with leading zeros assumed if necessary, are interpreted as the fractional part of the value
 represented. The string of digits may contain more digits than a processor uses to approximate the value. The
 basic form may be followed by an exponent of one of the following forms:
   • a sign followed by a digit-string;
   • the letter E followed by zero or more blanks, followed by a signed-digit-string;
   • the letter D followed by zero or more blanks, followed by a signed-digit-string.

If I am reading that correctly then
```
1234.567e89
1234.567E89
1234.567d89
1234.567D89
1234.567+89
```
should all be the same. So would adding a `+` after any of `e, E, d, or D` and likewise the negative exponent versions.

Reply:
Well... that's bizarre... I would never have guessed that was a permissible way of defining a float.  Kind of wish it weren't, but it doesn't prove your test is correct (and that the previous implementation was actually deficient).